traits/trait_types.py

```diff
@@ -1964,18 +1969,29 @@ def __init__(self, *args, **metadata):
                 )
         else:
             default_value = args[0]
-            if (len(args) == 1) and isinstance(default_value, EnumTypes):
-                args = default_value
-                default_value = enum_default(args)
-            elif (len(args) == 2) and isinstance(args[1], EnumTypes):
-                args = args[1]
+            if len(args) == 1:
+                self.values = args[0]
+                default_value = enum_default(self.values)
+
+                # If values is not a Collection, treat them differently
+                if isinstance(self.values, EnumTypes):
+                    self.values = tuple(self.values)
+                elif is_excluded_collection(self.values):
+                    self.values = {self.values}
+
+            elif len(args) == 2:
+                if is_excluded_collection(args[1]):
+                    self.values = tuple(args)
+                elif is_collection(args[1]):
+                    self.values = args[1]
```

Comment:
Please  could you add a test that exercises the case where neither the `if` nor the `elif` condition is true? (What is `self.values` in this case?)

Reply:
There is a bug here, `Enum(1,2,3)` wouldn't work since `self.values` will not be set. Made a fix and added the test.